lib/awful/spawn.lua

```diff
@@ -370,6 +370,18 @@ function spawn.easy_async(cmd, callback)
     })
 end
 
+--- Call spawn.easy_async with a shell.
+-- This calls `cmd` with `$SHELL -c` (via `awful.util.shell`).
+-- @tparam string|table cmd The command.
+-- @function callback The callback function.
+-- @see spawn.easy_async for details about `callback`.
+function spawn.easy_async_with_shell(cmd, callback)
+    if cmd and cmd ~= "" then
+        cmd = { util.shell, "-c", cmd }
+        return spawn.easy_async(cmd, callback)
+    end
+end
```

Comment:
I would remove the if and use `return spawn.easy_async(cmd or "", callback)`. The rational is that the return values should be consistent with `easy_async`.